File: src/test/java/com/soapdogg/lispInterpreter/parser/AtomNodeParserTest.java
package com.soapdogg.lispInterpreter.parser;

import com.soapdogg.lispInterpreter.datamodels.AtomNode;
import com.soapdogg.lispInterpreter.datamodels.ParserResult;
import com.soapdogg.lispInterpreter.datamodels.Token;
import com.soapdogg.lispInterpreter.generator.NodeGenerator;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mockito.Mockito;

import java.util.Collections;
import java.util.List;

public class AtomNodeParserTest {

    private List<Token> tokens;

    private ParserResult expected;

    private AtomNodeParser atomNodeParser;

    @BeforeEach
    void setup() {
        NodeGenerator nodeGenerator = Mockito.mock(NodeGenerator.class);
        ParserResultBuilder parserResultBuilder = Mockito.mock(ParserResultBuilder.class);

        String value = "value";
        Token headToken = Mockito.mock(Token.class);
        Mockito.when(headToken.getValue()).thenReturn(value);

        AtomNode atomNode = Mockito.mock(AtomNode.class);
        Mockito.when(nodeGenerator.generateAtomNode(value)).thenReturn(atomNode);

        tokens = Collections.singletonList(headToken);

        expected = Mockito.mock(ParserResult.class);
        Mockito.when(
            parserResultBuilder.buildParserResult(
                atomNode,
                Collections.emptyList()
            )
        ).thenReturn(expected);

        atomNodeParser = new AtomNodeParser(
            nodeGenerator,
            parserResultBuilder
        );
    }

    @Test
    void atomNodeParserTest() {
        ParserResult actual = atomNodeParser.parseAtomNode(tokens);

        Assertions.assertEquals(expected, actual);
    }
}

File: src/test/java/com/soapdogg/lispInterpreter/parser/AtomNodeParserTest.kt
package com.soapdogg.lispInterpreter.parser

import com.soapdogg.lispInterpreter.converter.NodeConverter
import com.soapdogg.lispInterpreter.datamodels.AtomNode
import com.soapdogg.lispInterpreter.datamodels.ParserResult
import com.soapdogg.lispInterpreter.datamodels.Token
import com.soapdogg.lispInterpreter.generator.NodeGenerator
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.mockito.Mockito

class AtomNodeParserTest {
    private val nodeGenerator = Mockito.mock(NodeGenerator::class.java)
    private val nodeConverter = Mockito.mock(NodeConverter::class.java)
    private val atomNodeParser = AtomNodeParser(
        nodeGenerator,
        nodeConverter
    )

    @Test
    fun atomNodeParserTest() {
        val value = "value"
        val headToken = Mockito.mock(Token::class.java)
        Mockito.`when`(headToken.value).thenReturn(value)

        val atomNode = Mockito.mock(AtomNode::class.java)
        Mockito.`when`(nodeGenerator.generateAtomNode(value)).thenReturn(atomNode)

        Mockito.`when`(nodeConverter.convertNodeV2ToNode(atomNode)).thenReturn(atomNode)

        val actual = atomNodeParser.parseAtomNode(headToken)
        Assertions.assertEquals(atomNode, actual)
    }
}